clicache/src/CacheTransactionManager.cpp

```diff
@@ -33,111 +33,165 @@ namespace Apache
 
       void CacheTransactionManager::Begin( )
       {
-        _GF_MG_EXCEPTION_TRY2
-
+        try {
           m_nativeptr->begin( );
-
-        _GF_MG_EXCEPTION_CATCH_ALL2
+        }
+        catch (const apache::geode::client::Exception& ex) {
+          throw Apache::Geode::Client::GeodeException::Get(ex);
+        }
+        catch (System::AccessViolationException^ ex) {
+          throw ex;
+        }
       }
 
       void CacheTransactionManager::Prepare( )
       {
-        _GF_MG_EXCEPTION_TRY2
-
+        try {
```

Comment:
Did you consider using lambdas to wrap the function body between these two macros to keep the common code non-repetitive?

Something like (untested):
```C++
void CacheTransactionManager::Prepare() {
  UnmanagedUtils::run([&](){
    m_nativeptr->prepare();
  });
}

...
template <typename F>
void NativeUtils::run(F function) {
  try {
    f();
  } catch (const apache::geode::client::Exception& ex) {
    throw Apache::Geode::Client::GeodeException::Get(ex);
  } catch (System::AccessViolationException^ ex) {
    throw ex;
  }
}
```

Reply:
Seems like a nice refactor.

Reply:
We don't currently use any lambdas in C++/CLI code. It's not clear to me if it is even supported. Possibly for unmanaged code I believe you are indicating.  However, for this PR I'm not sure it's with introducing new features in a deprecated compiler.